document-store/src/main/java/org/hypertrace/core/documentstore/postgres/query/v1/parser/filter/PostgresNotContainsRelationalFilterParser.java

```diff
@@ -1,19 +1,32 @@
 package org.hypertrace.core.documentstore.postgres.query.v1.parser.filter;
 
-import static org.hypertrace.core.documentstore.postgres.query.v1.parser.filter.PostgresContainsRelationalFilterParser.prepareJsonValueForContainsOp;
-
 import org.hypertrace.core.documentstore.expression.impl.RelationalExpression;
+import org.hypertrace.core.documentstore.postgres.query.v1.parser.filter.nonjson.field.PostgresContainsRelationalFilterParserNonJsonField;
 
 class PostgresNotContainsRelationalFilterParser implements PostgresRelationalFilterParser {
+  private static final PostgresContainsRelationalFilterParser jsonContainsParser =
+      new PostgresContainsRelationalFilterParser();
+  private static final PostgresContainsRelationalFilterParserNonJsonField nonJsonContainsParser =
+      new PostgresContainsRelationalFilterParserNonJsonField();
+
   @Override
   public String parse(
       final RelationalExpression expression, final PostgresRelationalFilterContext context) {
     final String parsedLhs = expression.getLhs().accept(context.lhsParser());
-    final Object parsedRhs = expression.getRhs().accept(context.rhsParser());
 
-    final Object convertedRhs = prepareJsonValueForContainsOp(parsedRhs);
-    context.getParamsBuilder().addObjectParam(convertedRhs);
+    String flatStructureCollection = context.getFlatStructureCollectionName();
+    boolean isFirstClassField =
+        flatStructureCollection != null
+            && flatStructureCollection.equals(context.getTableIdentifier().getTableName());
```

Comment:
It looks like we are applying to specific table name match. So, either table is flat table or it may contains json or non-json field. Looks like we are adding few checks, assuming we will do re-work?